ontology/coap.ttl

```diff
@@ -0,0 +1,165 @@
+                                    vann:preferredNamespacePrefix "cov" ;
+                                    vann:preferredNamespaceUri : ;
+                                    rdfs:comment "Vocabulary to represent CoAP messages in RDF."@en ;
+                                    rdfs:label "CoAP in RDF"@en .
+
+_:genid4 rdf:type schema:Organization ;
+          schema:name "Siemens AG" ;
+          schema:url "https://www.siemens.com/" .
+
+#################################################################
+#    Object Properties
+#################################################################
+
+
+###  https://www.w3.org/2019/wot/modbus#hasFunction
+:hasMethod rdf:type owl:ObjectProperty ;
+             rdfs:domain :Request ;
+             rdfs:comment "Indicates the CoAP method to be used in the request."@en ;
+             rdfs:isDefinedBy <> ;
+             rdfs:seeAlso "See the <a href=\"https://www.iana.org/assignments/core-parameters/#method-codes\">CoAP Method Codes</a> registry for the set of possible values.";
+             rdfs:label "hasMethod"@en .
+
+:hasBlockwise rdf:type owl:ObjectProperty ;
+        rdfs:domain :Message ;
+        rdfs:comment "Indicates that block-wise transfer [[RFC7959]] is supported."@en ;
+        rdfs:isDefinedBy <> ;
+        rdfs:label "hasBlockwise"@en .
+
+:hasQBlockwise rdf:type owl:ObjectProperty ;
+        rdfs:domain :Message ;
+        rdfs:comment "Indicates that quick block-wise transfer [[RFC9177]] is supported."@en ;
+        rdfs:isDefinedBy <> ;
+        rdfs:label "hasQBlockwise"@en .
+
+:hasHopLimit rdf:type owl:ObjectProperty ;
+        rdfs:domain :Request ;
+        rdfs:comment "Indicates that a CoAP Hop-Limit option [[RFC8768]] should be included in the request."@en ;
+        rdfs:isDefinedBy <> ;
+        rdfs:label "hasHopLimit"@en .
+
+:hasContentFormat rdf:type owl:ObjectProperty ;
+        rdfs:domain :Message ;
+        rdfs:comment "Indicates a numeric CoAP Content-Format [[RFC7252]] as a combination of Media Type and Content Coding. Must match the given <code>contentType</code> and <code>contentCoding</code> members."@en ;
+        rdfs:isDefinedBy <> ;
+        rdfs:label "hasContentFormat"@en .
+
+:hasAccept rdf:type owl:ObjectProperty ;
+        rdfs:domain :Request ;
+        rdfs:comment "Indicates that a Consumer must include an Accept option [[RFC7252]] with the given value when requesting the resource associated with this Form."@en ;
+        rdfs:isDefinedBy <> ;
+        rdfs:label "hasAccept"@en .
+
+#################################################################
+#    Data properties
+#################################################################
+
+:BlockWiseTransferParameters rdf:type owl:DatatypeProperty ;
+        rdfs:domain :Message ;
+        rdfs:isDefinedBy <> ;
+        rdfs:label "BlockWiseTransferParameters"@en .
+
+:hasblock2SZX rdf:type owl:DatatypeProperty ;
+        rdfs:domain :BlockWiseTransferParameters ;
+        rdfs:range :Blocksize ;
```

Comment:
The SHACL shape in [mapping.ttl](https://github.com/w3c/wot-binding-templates/pull/246/files#diff-bf9dcd7ed26a879e846cf4b2fbf9496df8e6585b146d5f3c3e4738e8926be483R80) requires that the value has datatype `xsd:unsignedShort`. So it seems either the SHACL shape is not satisfied when the value is an instance of `:Blocksize` or the OWL reasoner would try to make a block size like `"16"^^xsd:unsignedShort` an instance of `:Blocksize`, right?

Reply:
I was struggling a bit to reproduce the previous document state. Setting the datatype to `:Blocksize` in `mapping.ttl` replaces the `unsignedShort` with `Blocksize`, therefore I arrived at the current solution/workaround – but I suppose that the replacement would probably make more sense than mixing an enum data type with a "primitive" one, right?

Reply:
I think I'd remove the `:Blocksize` class and instances from coap.ttl and just use `xsd:unsignedShort`. After all, I don't see mapping.ttl doing any actual mapping; it's just validating that the triples in a TD have certain shape, similar to how the JSON Schema is validating that the JSON objects in the TD follow a certain schema.